lib/hackmode/p2p-backend-hackmode-peer.js

```diff
@@ -258,6 +258,10 @@ class P2pBackendHackmodePeer extends AsyncClass {
       throw new Error('cannot connect to peer without peerAddress ("a" param)')
     }
     const remotePeerAddress = uri.searchParams.get('a')
+    // Must be a different id
+    if (remotePeerAddress === this.getId()) {
```

Comment:
if it `throw`s an error, will it cause the process to crash?
Isn't it possible right now that nodes may be instructed to connect to themselves, in which case this WOULD result in the node crashing, which would be undesirable?

Reply:
Yes it will crash if there is no try/catch upstream.
I don't see a reason for a peer to connect to itself. `test_bin` tests all pass.
An early crash is more desirable than nothing at all and not understanding why peers are not succesfully talking to each other.

Reply:
I'm open to changing this to `console.error()` if you feel that its more appopriate.